inc/common/purge.php

```diff
@@ -196,6 +196,16 @@ function rocket_clean_post( $post_id, $post = null ) {
 		return false;
 	}
 
+	/**
+	 * Filters specific post_type cache from being cleared.
+	 *
+	 * @param bool $allow_exclusion Allow cache clearing for specific post type if true; otherwise if false.
+	 * @param string $post_type Post Type.
+	 */
+	if ( apply_filters( 'rocket_exclude_post_type_cache_clearing', false, $post->post_type ) ) {
+		return false;
+	}
```

Comment:
I think we could make this more generic, maybe `rocket_pre_clean_post`, passing the full `$post` value as an additional parameter. 

Something like the following:
```
$pre = apply_filters( 'rocket_pre_clean_post', null, $post );

if ( null !== $pre ) {
    return $pre;
}
```

That way it could be re-used for other cases, like the purge conditions we have below, or other compatibilities